0027-standardizing-comit-expiry-times.adoc

```diff
@@ -0,0 +1,133 @@
+
+We aim to produce a state diagram that fully describes a successful atomic swap.
+
+Initial state diagram:
+
+        `Started` --> `Alpha Funded` --> `Beta Funded` --> `Beta Redeemed` --> `Alpha Redeemed` --> `Completed`
+
+There exists some sequence of actions that enables a swap to transition to 'Completed'.
+We first annotate the state diagram with actions and then describe each action below.
+
+Annotated state diagram:
+
+        `Started` - fund - mine - finality -> `Alpha Funded` - fund - mine - finality -> `Beta Funded` - redeem - mine - finality -> `Beta Redeemed` - redeem - mine - finality -> `Alpha Redeemed` --> `Completed`
+
+We now define the term 'done' to mean the swap is successfully completed from the perspective of the actor.
+For Alice 'done' is defined as `Beta Redeemed`.
+For Bob 'done' is defined as `Alpha Redeemed`.
+We also note that while the state transitions are the same for both parties the actions required to transition are not the same for both parties.
+
+For each action that an actor must take there is some reasonable time period in which the action can be completed.
+In order to reason about the expiries each actor must know the actions they must take but also the actions the counterparty must take.
+This is because we wish to reason about the usefulness of the expiries to both ourselves and our counterparty.
+At any stage of the swap the sum of the actions to transition from the current state to done represents the time required to complete the swap.
+
+We will now describe each of the actions that appears in the annotated state diagram.
+
+* fund, redeem: Fund and redeem refer to creating the fund/redeem transaction and submitting it to the blockchain network.
+* mine: The time it takes for an unconfirmed transaction to be mined i.e., included in a block.
+* finality: The time it takes for a transaction, once mined, to reach finality.
+
+Each of these times is specific to the actor, the transaction type (redeem/refund), and the underlying blockchain.
+
+We now model the swap from each actors perspective.
+
+Annotated state diagram by actor:
+
+Alice:
+
+        `Started` - fund - mine - finality -> `Alpha Funded` - fund - mine - finality -> `Beta Funded` - redeem - mine - finality -> `Beta Redeemed` --> `Done`
+
+Bob:
+
+        `Started` - fund - mine - finality -> `Alpha Funded` - fund - mine - finality -> `Beta Funded` - redeem(beta) - redeem(alpha) - mine - finality -> `Done`
+
+Here we see that as soon as the beta redeem transaction becomes known to the network Bob can create his redeem transaction and submit it to the alpha blockchain network.
+
+
+== Transition Period per Action
+
+=== Fund/Redeem
+
+There exists some amount of time that it takes for an actor to create a transaction and submit it to the network.
+Of consideration some transactions may be created in software some may take user intervention.
```

Comment:
This sentence has a bit of a weird structure to me.

Also, the word "intervention" for me has the co-notation of "preventing" something to happen. "Interaction" might be better.

Reply:
Yep, nice. 'interaction' is better. Thanks